WeatherService.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Weather_Monitoring_System.Bots;
+using Weather_Monitoring_System.DataParsers;
+using Weather_Monitoring_System.Models;
+
+namespace Weather_Monitoring_System
+{
+    public class WeatherService
+    {
+        private readonly IWeatherDataParser _dataParser;
+
+        private readonly List<Bot> _botServices = [new RainBotService(), new SnowBotService(), new SunBotService()];
```

Comment:
It's really bad to get instances of this service in this way, this way you will make this service coupled with the bot's services. Please apply the IoC principle here.